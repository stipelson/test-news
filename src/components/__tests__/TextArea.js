import React from 'react';
import { create, act } from 'react-test-renderer';

import TextArea from '../TextArea';

describe('TextArea', () => {
  it('renders textarea correctly', () => {
    const tree = create(
      <TextArea
        label="Message"
        rows="4"
        name="message"
        onChange={() => {
          return false;
        }}
        value=""
        errorMessage={undefined}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders an input textarea by default', () => {
    const tree = create(<TextArea />);
    const instance = tree.root;
    const component = instance.findByType('textarea');
    expect(component.props.className).toEqual('custom-textarea');
  });

  it('renders an input textarea by default', () => {
    const tree = create(<TextArea />);
    const instance = tree.root;
    const component = instance.findByType('textarea');
    expect(component.props.className).toEqual('custom-textarea');
  });

  it('TextArea test onChange', () => {
    const onChangeMock = jest.fn();
    const event = {
      target: { value: 'This is just for test' },
    };
    let component;

    act(() => {
      component = create(
        <TextArea
          label="Message"
          rows="4"
          name="message"
          onChange={onChangeMock}
          value="Pre value"
          errorMessage={undefined}
        />
      );
    });

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    let instance = component.root;

    // If no exist textarea get error
    const textArea = instance.findByType('textarea');

    // Value is passed down to the input
    expect(textArea.props.value).toEqual('Pre value');

    // manually trigger the onchange
    textArea.props.onChange(event);
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually change value
    // update with some different props
    act(() => {
      component.update(
        <TextArea
          label="Message"
          rows="4"
          name="message"
          onChange={onChangeMock}
          value={event.target.value}
          errorMessage={undefined}
        />
      );
    });
    // make assertions on root
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('Change value of textarea manually', () => {
    const onChangeMock = jest.fn();
    const event = {
      target: { value: 'This is just for test' },
    };
    let component;

    act(() => {
      component = create(
        <TextArea
          label="Message"
          rows="4"
          name="message"
          onChange={onChangeMock}
          value="Pre value"
          errorMessage={undefined}
        />
      );
    });

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually change value
    // update with some different props
    act(() => {
      component.update(
        <TextArea
          label="Message"
          rows="4"
          name="message"
          onChange={onChangeMock}
          value={event.target.value}
          errorMessage={undefined}
        />
      );
    });
    // make assertions on root
    expect(component.toJSON()).toMatchSnapshot();
  });
});
