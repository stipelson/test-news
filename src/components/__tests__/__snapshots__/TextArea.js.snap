// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea renders an input default textarea 1`] = `
<div
  className="eui-text-field  "
>
  <div
    className="eui-text-field-wrapper"
  >
    <textarea
      aria-errormessage="undefinedErrorMessage"
      aria-invalid={false}
      className="custom-textarea"
      role="textbox"
    />
    <div
      className="eui-text-field-footer"
    >
      <span
        className="eui-text-field-message eui-text-field-error-message "
        id="undefinedErrorMessage"
        role="alert"
      />
      <span
        className="eui-text-field-message eui-text-field-help-text"
      />
    </div>
  </div>
  <label />
</div>
`;

exports[`TextArea renders textarea with data correctly 1`] = `
<div
  className="eui-text-field  "
>
  <div
    className="eui-text-field-wrapper"
  >
    <textarea
      aria-errormessage="undefinedErrorMessage"
      aria-invalid={false}
      aria-label="Message"
      className="custom-textarea"
      name="message"
      onChange={[Function]}
      role="textbox"
      rows="4"
      value=""
    />
    <div
      className="eui-text-field-footer"
    >
      <span
        className="eui-text-field-message eui-text-field-error-message "
        id="undefinedErrorMessage"
        role="alert"
      />
      <span
        className="eui-text-field-message eui-text-field-help-text"
      />
    </div>
  </div>
  <label>
    Message
  </label>
</div>
`;
